bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitValveImpl.java

```diff
@@ -73,6 +78,73 @@ public HomekitValveImpl(HomekitTaggedItem taggedItem, List<HomekitTaggedItem> ma
         this.activeReader = new BooleanItemReader(getItem(HomekitCharacteristicType.ACTIVE_STATUS, GenericItem.class),
                 OnOffType.ON, OpenClosedType.OPEN);
         getServices().add(new ValveService(this));
+        final String timerConfig = getAccessoryConfiguration(CONFIG_TIMER, "");
+        homekitTimer = timerConfig.equalsIgnoreCase("yes") || timerConfig.equalsIgnoreCase("true");
+        if (homekitTimer) {
+            addRemainingDurationCharacteristic(taggedItem, updater);
+        }
+    }
+
+    private void addRemainingDurationCharacteristic(HomekitTaggedItem taggedItem, HomekitAccessoryUpdater updater) {
+        logger.trace("addRemainingDurationCharacteristic for {}", taggedItem);
+        ((ValveService) getPrimaryService()).addOptionalCharacteristic(new RemainingDurationCharacteristic(() -> {
+            int remainingTime = 0;
+            ScheduledFuture<?> future = valveTimer;
+            if (future != null && !future.isDone()) {
+                remainingTime = java.lang.Math.toIntExact(future.getDelay(TimeUnit.SECONDS));
+            }
+            return CompletableFuture.completedFuture(remainingTime);
+        }, (callback) -> updater.subscribe((GenericItem) taggedItem.getItem(), REMAINING_DURATION.getTag(), callback),
+                () -> updater.unsubscribe((GenericItem) taggedItem.getItem(), REMAINING_DURATION.getTag())
```

Comment:
Couldn't getSubscriber and getUnsubscriber be used here?

Reply:
good one. missed that one.